src/main/java/uk/gov/ons/census/fwmt/jobservice/transition/Transitioner.java

```diff
@@ -0,0 +1,199 @@
+            messageCache.toBuilder().messageType(returnedRules[1]).message(message).build());
+      }
+
+      if (storedCacheType.equals("EMPTY")) {
+        eventManager
+            .triggerEvent(String.valueOf(rmRequest.getCaseId()), RM_CANCEL_REQUEST_STORED,
+                "TM Action", "Update received before create case. Update has been stored.");
+      }
+    }
+  }
+
+  public void processCreateOrUpdateTransition(GatewayCache cache, FwmtActionInstruction rmRequest,
+        List<InboundProcessor<FwmtActionInstruction>> processors) throws GatewayException {
+      boolean noAction = false;
+      boolean cacheRmRequest;
+      String message;
+      String actionInstruction;
+      ObjectMapper rmRequestMapper = new ObjectMapper();
+
+      actionInstruction = rmRequest.getActionInstruction().toString();
+
+      String[] returnedRules = setTransitionRules(cache, actionInstruction, rmRequest.getCaseId());
+
+      cacheRmRequest = Boolean.parseBoolean(returnedRules[2]);
+
+      switch (returnedRules[0]) {
+      case "NO_ACTION":
+        eventManager
+            .triggerEvent(rmRequest.getCaseId(), NO_ACTION_REQUIRED,
+                "Case Ref", rmRequest.getCaseRef());
+        noAction = true;
+        break;
+      case "REJECT":
+        eventManager.triggerErrorEvent(this.getClass(), "Request from RM rejected",
+            String.valueOf(rmRequest.getCaseId()), ROUTING_FAILED);
+        break;
+      case "PROCESS":
+        processors.get(0).process(rmRequest, cache);
+        break;
+      case "MERGE":
+        //TODO - will make a call to mergeRecords() once code is complete
+        break;
+      default:
+        break;
+      }
+
+    MessageCache messageCache = messageCacheService.getById(rmRequest.getCaseId());
+
+    if (!cacheRmRequest && messageCache != null) {
+        messageCacheService.deleteByCaseId(rmRequest.getCaseId());
+      }
+
+      if (noAction && cacheRmRequest) {
+        try {
+          message = rmRequestMapper.writeValueAsString(rmRequest);
+        } catch (JsonProcessingException processFailure) {
+          return;
+        }
+
+        if (messageCache == null) {
+          messageCacheService.save(
+              MessageCache.builder().caseId(rmRequest.getCaseId()).messageType(returnedRules[1]).message(message)
+                  .build());
+        } else {
+          messageCacheService.save(
+              messageCache.toBuilder().messageType(returnedRules[1]).message(message).build());
+        }
+      }
+      if (actionInstruction.equals("UPDATE") && storedCacheType.equals("EMPTY")) {
+        eventManager
+            .triggerEvent(String.valueOf(rmRequest.getCaseId()), RM_UPDATE_REQUEST_STORED,
+                "Case Ref", rmRequest.getCaseRef(),
+                "TM Action", "Update received before create case. Update has been stored.");
+      }
+  }
+
+  public String[] setTransitionRules(GatewayCache cache, String actionRequest, String caseId) throws GatewayException {
```

Comment:
change the name to something like collect... as youre not really setting

Reply:
This is true. Awful at naming! Just changed it now.